robot/web/web.go

```diff
@@ -984,6 +984,8 @@ func (svc *webService) foreignServiceHandler(srv interface{}, stream googlegrpc.
 			defer wg.Done()
 
 			var err error
+			// process first message before waiting for more messages
+			err = bidiStream.SendMsg(firstMsg)
```

Comment:
Does the foreign service handler essentially open a bidirectional stream "between" the client and the foreign service and attempt to pipe messages back and forth between the two? And, was the bug that we would receive the first message from the client but not actually forward it through the created bidi stream?

Reply:
yep, it basically does what a regular service handler does but without knowing the underlying types upfront.

ya, it seems like we just forget about the first message. Somewhat unsure if that was intentional (since reproducing this wasn't too hard), but also I don't see a reason why we would've wanted to do that. @Otterverse do you know if the previous behavior was intentional?

Reply:
I can't imagine it was intentional. Also surprised it hid in plain sight this long. Good catch.